Authentication does not seem to be working quite right

First Off, awesome tool...the amount of scripting that I can replace with this tool is amazing!

When I try to build an image that requires one of our private images as a base image the authentication is not working:

```bash
$ docker login -u <my-user> -p <my-pass> <my-url>
Login Succeeded
$ dobi
Step 1/10 : FROM <company>/<project>/alpine-docker:17.03.0-ce
[ERROR] failed to execute task "<task>:build": Get <image-url>: unknown: Authentication is required
$ docker pull <company>/<project>/alpine-docker:17.03.0-ce
17.03.0-ce: Pulling from <project>/alpine-docker
ec37562cf8fa: Already exists 
371cc900ff27: Pull complete 
51ad1472b975: Pull complete 
76c902807740: Pull complete 
Digest: sha256:c5ca1b30d6731ac39df368936b03a83824be99d98fb202a12beb55ad7b4ac541
Status: Downloaded newer image for <company>/<project>/alpine-docker:17.03.0-ce
````

dobi.yaml
```yaml
meta:
  project: <project name>
  default: build

image=<image name>:
  image: <company>/<project>/<image name>
  context: <directory>
  tags:
    - 'latest'
    - '{git.short-sha}'
  description: "Build the <image name> image"

alias=build:
  tasks: ['<image name>:build']
  description: "Build images"

alias=tag:
  tasks: ['<image name>:tag']
  description: "Build and tag images"

alias=push:
  tasks: ['<image name>:push']
  description: "Build, tag, and push images"

alias=clean:
  tasks: ['<image name>:rm']
  description: "Remove all the images"
```

My question is why is dobi not using the same auth mechanism as docker? And if it is then it doesn't seem to be exactly the same.

--- a/tasks/context/execcontext.go
+++ b/tasks/context/execcontext.go
@@ -48,6 +48,16 @@ func (ctx *ExecuteContext) GetAuthConfig(repo string) docker.AuthConfiguration {
 	return auth
 }
 
+// GetAuthConfigs returns all the authorization configs in the config file. This
+// is used by build, because the repo isn't known until after the Dockerfile is
+// parsed.
+func (ctx *ExecuteContext) GetAuthConfigs() docker.AuthConfigurations {
+	if ctx.authConfigs == nil {
+		return docker.AuthConfigurations{}
+	}
+	return *ctx.authConfigs
+}
+
 // NewExecuteContext craetes a new empty ExecuteContext
 func NewExecuteContext(
 	config *config.Config,

--- a/tasks/image/build.go
+++ b/tasks/image/build.go
@@ -86,6 +86,7 @@ func buildImage(ctx *context.ExecuteContext, t *Task) error {
 			OutputStream:   out,
 			RawJSONStream:  true,
 			SuppressOutput: ctx.Quiet,
+			AuthConfigs:    ctx.GetAuthConfigs(),
 		})
 	}); err != nil {
 		return err

--- a/tasks/image/utils_test.go
+++ b/tasks/image/utils_test.go
@@ -19,7 +19,7 @@ func TestParseAuthRepoPrivateRepoAndPort(t *testing.T) {
 }
 
 func TestParseAuthRepoPrivateRepo(t *testing.T) {
-	repo, err := parseAuthRepo("myrepo.net/dnephin/foo")
+	repo, err := parseAuthRepo("myrepo.net/dnephin/foo:tag")
 	assert.Nil(t, err)
 	assert.Equal(t, repo, "myrepo.net")
 }

